src/main/java/io/github/genomicdatainfrastructure/discovery/filters/infrastructure/ckan/CkanFilterBuilder.java

```diff
@@ -257,22 +264,21 @@ private Map<String, FilterMetadata> extractFiltersMetadata(DatasetsConfig datase
                 .flatMap(group -> ofNullable(group.filters()).orElseGet(Set::of)
                         .stream()
                         .map(filter -> Map.entry(filter.key(),
-                                new FilterMetadata(Boolean.TRUE.equals(filter.isDateTime()),
-                                        Boolean.TRUE.equals(filter.isNumber()), group.key()))))
+                                new FilterMetadata(Optional.ofNullable(filter.type())
+                                        .orElse(DEFAULT_FILTER_TYPE),
+                                        group.key()))))
                 .collect(LinkedHashMap::new,
                         (map, entry) -> map.put(entry.getKey(), entry.getValue()),
                         Map::putAll);
     }
 
     private static final class FilterMetadata {
 
-        private final boolean isDateTime;
-        private final boolean isNumber;
+        private final FilterType type;
         private final String group;
 
-        private FilterMetadata(boolean isDateTime, boolean isNumber, String group) {
-            this.isDateTime = isDateTime;
-            this.isNumber = isNumber;
+        private FilterMetadata(FilterType type, String group) {
+            this.type = type;
             this.group = group;
         }
```

Comment:
This could be a record to reduce the number of lines.

```suggestion
    private record FilterMetadata(FilterType type, String group) {}
```

(but all callers need to be updated for this to work)